### PR TITLE
Get only bs prefixed data attributes

### PR DESCRIPTION
**Pull request by rohit2sharma95**
https://github.com/twbs/bootstrap/pull/31827#issuecomment-724814337

### CODE REVIEW

File: js/src/dom/manipulator.js [hunk 1]
```diff
@@ -43,19 +43,15 @@ const Manipulator = {
...
-    const attributes = {
-      ...element.dataset
-    }
-
-    Object.keys(attributes)
+    return Object.keys(element.dataset)
       .filter(key => key.startsWith('bs'))
-      .forEach(key => {
+      // eslint-disable-next-line unicorn/no-reduce
+      .reduce((attributes, key) => {
```
Comment (XhmikosR) [inline js/src/dom/manipulator.js:49]: How much worse would this be in terms of readability if you used a for or forEach loop? We don't use reduce anywhere else and while I prefer readability, reduce is a special case.

Comment (rohit2sharma95) [reply js/src/dom/manipulator.js:49]: Okay understood that's why you are not using `reduce`, I was thinking about it already. Let me update it to `forEach`.

Comment (XhmikosR) [reply js/src/dom/manipulator.js:49]: Thanks! Just thinking out loud, we might go with your original solution anyway :)

Comment (rohit2sharma95) [reply js/src/dom/manipulator.js]: Updated 👍

Comment (XhmikosR) [reply js/src/dom/manipulator.js]: Thanks! I say we go with this solution and when we switch the whole data file to use a `Map` we can proceed with any further cleanup. Patches welcome as usual, after we land this :)

**Comment by rohit2sharma95** [comment]
@XhmikosR It means I got approval from your end 🙂
Should I merge this PR now?

**Comment by XhmikosR** [comment]
Yup. I'll probably rebase the branch before the final merge, though :)

**Pull request merged**